samples/Meeting/src/app/App.tsx

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { GroupCallLocator, GroupLocator, TeamsMeetingLinkLocator } from '@azure/communication-calling';
+import { CommunicationUserIdentifier } from '@azure/communication-common';
+import { initializeIcons, Spinner } from '@fluentui/react';
+import React, { useEffect, useState } from 'react';
+import {
+  buildTime,
+  callingSDKVersion,
+  createGroupId,
+  fetchTokenResponse,
+  getGroupIdFromUrl,
+  getTeamsLinkFromUrl,
+  isMobileSession,
+  isOnIphoneAndNotSafari,
+  isSmallScreen,
+  navigateToHomePage
+} from './utils/AppUtils';
+import { CallError } from './views/CallError';
+import { MeetingScreen } from './views/MeetingScreen';
+import { EndCall } from './views/EndCall';
+import { HomeScreen } from './views/HomeScreen';
+import { UnsupportedBrowserPage } from './views/UnsupportedBrowserPage';
+import { getThreadId } from './utils/getThreadId';
+import { createThread } from './utils/createThread';
+import { checkThreadValid } from './utils/checkThreadValid';
+import { getEndpointUrl } from './utils/getEndpointUrl';
+import { joinThread } from './utils/joinThread';
+
+console.log(
+  `ACS sample Meeting app. Last Updated ${buildTime} Using @azure/communication-calling:${callingSDKVersion}`
+);
+
+initializeIcons();
+
+type AppPages = 'home' | 'meeting' | 'endMeeting';
+
+const webAppTitle = document.title;
+
+const App = (): JSX.Element => {
+  const [page, setPage] = useState<AppPages>('home');
+
+  // User credentials to join a call with - these are retrieved from the server
+  const [token, setToken] = useState<string>();
+  const [userId, setCallUserId] = useState<CommunicationUserIdentifier>();
+  const [userCredentialFetchError, setUserCredentialFetchError] = useState<boolean>(false);
+
+  // Call details to join a call - these are collected from the user on the home screen
+  const [callLocator, setCallLocator] = useState<GroupLocator | TeamsMeetingLinkLocator>(createGroupId());
+  const [displayName, setDisplayName] = useState<string>('');
+
+  // Chat endpoint and thread id
+  const [endpointUrl, setEndpointUrl] = useState('');
+  const [threadId, setThreadId] = useState('');
+
+  // Get Azure Communications Service token from the server
+  useEffect(() => {
+    (async () => {
+      try {
+        const { token, user } = await fetchTokenResponse();
+        setToken(token);
+        setCallUserId(user);
+      } catch (e) {
+        console.error(e);
+        setUserCredentialFetchError(true);
+      }
+    })();
+  }, []);
+
+  const ERROR_TEXT_THREAD_NOT_RECORDED = 'Thread id is not recorded in server';
+  const ALERT_TEXT_TRY_AGAIN = "You can't be added at this moment. Please wait at least 60 seconds to try again.";
+
+  const onCreateThread = async (): Promise<string> => {
```

Comment:
nit: remove the `on` here as this isn't a callback, suggestion rename to something like `getThread()`

Reply:
This is a utility function, to shorten this large function and make it more readable, hoist this either to its own file in utils/ or hoist out the function and move to the bottom of this file.

Reply:
Recommend moving this to beside `getGroupIdFromUrl`

Reply:
Recommend moving to where `getGroupIdFromUrl` is defined